File: simple-log-processor/src/test/java/com/don11995/log/LogSettings.java
/*
 * Modified by Vladyslav Lozytskyi on 12.02.18 13:09
 * Copyright (c) 2018. All rights reserved.
 */

package com.don11995.log;

public interface LogSettings {

    @MapClassInner(methods = "innerTest", prefixes = "TEST")
    void mapClassToTest(ClassToTest classToTest);

    @MapFieldInner(method = "audioFocus", names = {
            "AUDIOFOCUS_GAIN",
            "AUDIOFOCUS_GAIN_TRANSIENT",
            "AUDIOFOCUS_LOSS",
            "AUDIOFOCUS_TRANSIENT",
            "AUDIOFOCUS_TRANSIENT_CAN_DUCK",
    })
    void audioFocus(Constants audioManager);
}

File: simple-log-processor/src/test/java/com/don11995/log/LogSettings.kt
/*
 * Modified by Vladyslav Lozytskyi on 05.05.18 16:50
 * Copyright (c) 2018. All rights reserved.
 */

package com.don11995.log

@Suppress("unused")
interface LogSettings {

    @MapClassInner(methods = ["innerTest"], prefixes = ["TEST"])
    fun mapClassToTest(classToTest: ClassToTest)

    @MapFieldInner(method = "audioFocus", names = ["AUDIOFOCUS_GAIN", "AUDIOFOCUS_GAIN_TRANSIENT", "AUDIOFOCUS_LOSS", "AUDIOFOCUS_TRANSIENT", "AUDIOFOCUS_TRANSIENT_CAN_DUCK"])
    fun audioFocus(audioManager: Constants)
}
